immutablecollections/_immutabledict.py

```diff
@@ -23,18 +23,16 @@
 # cannot share type variables between outer and inner classes
 KT2 = TypeVar("KT2")
 VT2 = TypeVar("VT2")
-IT2 = Tuple[KT, VT]
+IT2 = Tuple[KT2, VT2]
 
 SelfType = TypeVar("SelfType")  # pylint:disable=invalid-name
 
-AllowableDictType = Union[  # pylint:disable=invalid-name
-    "ImmutableDict[KT, VT]", Dict[KT, VT]
-]
+AllowableDictType = Union[Iterable[IT], Mapping[KT, VT], "ImmutableDict[KT, VT]"]
```

Comment:
Since this no longer only includes dicts/mappings, how about `AllowableSourceType` or something like that?

Reply:
I approve in general but agree with @ConstantineLignos  re: the rename

Reply:
`AllowableSourceType` is looking too vague for me. How about `AllowableDictInputSourceType`?

Reply:
hmm - `AllowableSourceType` seems clear to me in the context in which it is used, but the longer form is fine if you prefer it

Reply:
It's because of usage for our `_to_immutabledict` converter that should go away eventually anyway, and that requires an import of `immutablecollections._immutabledict`. So on second thought, it doesn't need to be changed, since no one else should be using this.